File: core/src/main/java/pcb/uwu/unit/derived/area/AreaUnit.java
package pcb.uwu.unit.derived.area;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.unit.base.LengthUnit;

public class AreaUnit extends CompositeUnit {

	public AreaUnit(LengthUnit length, LengthUnit height) {
		super(new UnitCounter()
				.major(length)
				.major(height));
	}
}

File: core/src/main/java/pcb/uwu/unit/derived/fundamental/AreaUnit.kt
package pcb.uwu.unit.derived.fundamental

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.LuminousIntensityUnit
import pcb.uwu.unit.derived.electromagnetism.MagneticFieldUnit
import pcb.uwu.unit.derived.electromagnetism.MagneticFluxUnit
import pcb.uwu.unit.derived.optics.LuminanceUnit

open class AreaUnit : CompositeUnit
{
    constructor(length: LengthUnit, height: LengthUnit)
            : super(UnitCounter()
                        .major(length)
                        .major(height))

    constructor(luminousIntensityUnit: LuminousIntensityUnit,
                luminanceUnit: LuminanceUnit)
            : super(UnitCounter()
                        .major(luminousIntensityUnit)
                        .minor(luminanceUnit))

    constructor(magneticFluxUnit: MagneticFluxUnit,
                magneticFieldUnit: MagneticFieldUnit)
            : super(UnitCounter()
                        .major(magneticFluxUnit)
                        .minor(magneticFieldUnit))
}